lib/xctest.js

```diff
@@ -0,0 +1,180 @@
+import { startHouseArrestService, startTestmanagerdService, startInstallationProxyService, startInstrumentService } from './services';
+import { INSTRUMENT_CHANNEL } from './instrument';
+import { TESTMANAGERD_CHANNEL } from './testmanagerd';
+import { DTXMessageAuxBuffer } from './instrument/headers';
+import { NSUUID, XCTestConfiguration } from './instrument/transformer/nskeyed';
+import { uuidV4 } from '@appium/support/build/lib/util';
+import { getOSVersion } from './utilities';
+import log from './logger';
+import B from 'bluebird';
+
+class Xctest {
+    constructor(udid, xctestBundleId, targetBundleId = undefined) {
+        this.udid = udid;
+        this.running = false;
+        this._executing = false;
+        this.xctestBundleId = xctestBundleId;
+        this.targetBundleId = targetBundleId;
+    }
+
+    async _launchAppRunner(majorVersion, sessionIdentifier) {
+        const { PROCESS_CONTROL } = INSTRUMENT_CHANNEL;
+        const installation = await startInstallationProxyService(this.udid);
+        const lookupResult = await installation.lookupApplications({ bundleIds: [this.xctestBundleId] });
+        installation.close();
```

Comment:
Safe to close in `finally` (since the lookup could raise an error)?
```suggestion
        const installation = await startInstallationProxyService(this.udid);
        let lookupResult;
        try {
            lookupResult = await installation.lookupApplications({ bundleIds: [this.xctestBundleId] });
        } finally {
            installation.close();
        }
```

Reply:
Done